actioneval/filesearch.go

```diff
@@ -0,0 +1,296 @@
+// Copyright © 2019 Aqua Security Software Ltd. <info@aquasec.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actioneval
+
+import (
+	"archive/tar"
+	"fmt"
+	"github.com/aquasecurity/bench-common/util"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"gopkg.in/yaml.v2"
+)
+
+type FileSearchFilter struct {
+
+	// location where to start the search
+	searchLocation string
+	// file name filter
+	filter string
+	// file name filter pattern values contains
+	filterType util.TextFilterType
+
+	//file type filter dir or symbolic link
+	fileType util.FileFilterType
+	// permission 0600,0777,/777 ,-666 etc...
+	perm  int64
+	sMode util.PermissionSearchMode
+
+	tarHeaders []tar.Header
+
+	groupId int64
```

Comment:
I think you might be better off defining these as uint32 to avoid casting later on, and have 0 represent "not defined"